pkg/controller/common/util.go

```diff
@@ -1216,3 +1250,143 @@ func IsWaitForFirstConsumerEnabled(obj metav1.Object, gates featuregates.Feature
+	// the `image` has to be aligned correctly, so the space requested has to be aligned to
+	// next value that is a multiple of a block size
+	alignedSize := util.RoundUp(requestedSpace, util.DefaultAlignBlockSize)
+
+	// count overhead as a percentage of the whole/new size, including aligned image
+	// and the space required by filesystem metadata
+	spaceWithOverhead := int64(math.Ceil(float64(alignedSize) / (1 - filesystemOverhead)))
+	return spaceWithOverhead
+}
+
+// InflateSizeWithOverhead inflates a storage size with proper overhead calculations
+func InflateSizeWithOverhead(c client.Client, imgSize int64, pvcSpec *v1.PersistentVolumeClaimSpec) (resource.Quantity, error) {
+	var returnSize resource.Quantity
+
+	if util.ResolveVolumeMode(pvcSpec.VolumeMode) == v1.PersistentVolumeFilesystem {
+		fsOverhead, err := GetFilesystemOverheadForStorageClass(c, pvcSpec.StorageClassName)
+		if err != nil {
+			return resource.Quantity{}, err
+		}
+		// Parse filesystem overhead (percentage) into a 64-bit float
+		fsOverheadFloat, _ := strconv.ParseFloat(string(fsOverhead), 64)
+
+		// Merge the previous values into a 'resource.Quantity' struct
+		requiredSpace := GetRequiredSpace(fsOverheadFloat, imgSize)
+		returnSize = *resource.NewScaledQuantity(requiredSpace, 0)
+	} else {
+		// Inflation is not needed with 'Block' mode
+		returnSize = *resource.NewScaledQuantity(imgSize, 0)
+	}
+
+	return returnSize, nil
+}
+
+// IsUnbound returns if the pvc is not bound yet
+func IsUnbound(pvc *v1.PersistentVolumeClaim) bool {
+	return pvc.Spec.VolumeName == ""
+}
+
+// HasAnnOwnedByDataVolume returns if the obj has annotation
+// it is owned by DataVolume
+func HasAnnOwnedByDataVolume(obj metav1.Object) bool {
+	_, ok := obj.GetAnnotations()[AnnOwnedByDataVolume]
+	return ok
+}
+
+// IsImageStream returns true if registry source is ImageStream
+func IsImageStream(pvc *corev1.PersistentVolumeClaim) bool {
+	return pvc.Annotations[AnnRegistryImageStream] == "true"
+}
+
+// ShouldIgnorePod checks if a pod should be ignored.
+// If this is a completed pod that was used for one checkpoint of a multi-stage import, it
+// should be ignored by pod lookups as long as the retainAfterCompletion annotation is set.
+func ShouldIgnorePod(pod *corev1.Pod, pvc *corev1.PersistentVolumeClaim) bool {
+	retain := pvc.ObjectMeta.Annotations[AnnPodRetainAfterCompletion]
+	checkpoint := pvc.ObjectMeta.Annotations[AnnCurrentCheckpoint]
+	if checkpoint != "" && pod.Status.Phase == corev1.PodSucceeded {
+		return retain == "true"
+	}
+	return false
+}
+
+// BuildHTTPClient generates an http client that accepts any certificate, since we are using
+// it to get prometheus data it doesn't matter if someone can intercept the data. Once we have
+// a mechanism to properly sign the server, we can update this method to get a proper client.
+func BuildHTTPClient(httpClient *http.Client) *http.Client {
+	if httpClient == nil {
```

Comment:
Yeah I'll take some time to improve this. I added the argument since we use a global `httpClient` in the dv controller and I didn't want to mess anything with moving the function to the common code (can't just use global variables as we were doing).